cmake/modules/FindSaxon.cmake

```diff
@@ -50,11 +50,13 @@ if (JAVA_RUNTIME)
   # Find Saxon 6.5.x
   #
   find_file (SAXON
-    NAMES saxon.jar saxon6.jar saxon-6.5.5.jar saxon-6.5.4.jar saxon-6.5.3.jar
+    NAMES saxon.jar saxon6.jar saxon-6.5.5.jar saxon-6.5.4.jar saxon-6.5.3.jar saxon9he.jar
     PATH_SUFFIXES share/java
                   share/java/saxon
                   share/java/saxon6
-                  share/saxon-6.5/lib 
+                  share/saxon-6.5/lib
+                  java/saxon
+    PATHS ENV HOME
```

Comment:
We already use a dedicated environment variable in some places (FLICKCURL, LCMS2, LensFun and OpenEXR) so just adding that should do the trick.